api/py/ai/chronon/repo/validator.py

```diff
@@ -74,11 +74,27 @@ def is_identifier(s: str) -> bool:
     return re.fullmatch(identifier_regex, s) is not None
 
 
-def get_pre_derived_group_by_columns(group_by: GroupBy) -> List[str]:
+def get_pre_derived_group_by_features(group_by: GroupBy) -> List[str]:
     output_columns = []
+    # For group_bys with aggregations, aggregated columns
     if group_by.aggregations:
         for agg in group_by.aggregations:
             output_columns.extend(get_output_col_names(agg))
+    # For group_bys without aggregations, selected fields from query
+    else:
+        for source in group_by.sources:
+            if source.events:
+                output_columns.extend(source.events.query.selects.keys())
+            elif source.entities:
+                output_columns.extend(source.entities.query.selects.keys())
+            elif source.joinSource:
+                output_columns.extend(source.joinSource.query.selects.keys())
```

Comment:
as mentioned below, it can use `get_pre_derived_join_keys()` in this way:
```
for source in group_by.sources:
    output_columns.extend(get_pre_derived_join_keys(source))
```

Reply:
Updated